src/services/timerFunction.ts

```diff
@@ -0,0 +1,214 @@
+    execution.currentState === ExecutionStatus.Initialized
+  ) {
+    if (execution.currentState === ExecutionStatus.Initialized) {
+      execution = addElapsedTime(execution);
+    }
+    execution.currentState = ExecutionStatus.Finished;
+    await removeOngoingExecution(id);
+    let newFinishedExecution: FinishedExecution = {
+      activity: execution.activity,
+      role: execution.role,
+      date: execution.startTime,
+      duration: execution.elapsedTime,
+      currentState: execution.currentState,
+    };
+
+    return addFinishedExecution(newFinishedExecution);
+  }
+  console.error("Execução nunca foi iniciada.");
+  await removeOngoingExecution(id);
+  return false;
+};
+
+/** Função responsável por atualizar contagem de toda a lista de execuções correntes.
+ * Retorna false se não há execuções para serem atualizadas e true se há
+ */
+export const updateAll = async () => {
+  let strArray = await getOngoingExecutions();
+  if (!strArray || strArray.length === 0) {
+    console.warn("Não há execuções.");
+    return false;
+  }
+
+  let arExecutions: OngoingExecution[] = JSON.parse(strArray);
+  let execution: OngoingExecution;
+  let isOneRunning = false;
+  for (let i = 0; i < arExecutions.length; i++) {
+    execution = arExecutions[i];
+    if (execution.currentState === ExecutionStatus.Initialized) {
+      execution = addElapsedTime(execution);
+      isOneRunning = true;
+      setOngoingExecution(execution, i);
+    }
+  }
+  return isOneRunning;
+};
+
+/** Recebe tempo em segundos e devolve string formatada corretamente.
+ */
+export const timeToString = (time: number) => {
+  let strTime = moment.unix(time).utc().format("HH:mm:ss");
+  console.log(strTime);
+  return strTime;
+};
+
+const addElapsedTime = (execution: OngoingExecution) => {
+  let pauseTime: Moment = moment();
+  console.log(
+    "timerFunction.addElapsedTime: Horário atual: ",
+    pauseTime.format("YYYY-MM-DDTHH:mm:ss[Z]ZZ")
+  );
+  console.log(
+    "timerFunction.addElapsedTime: Contagem até este momento: ",
+    execution.elapsedTime
+  );
+
+  let diffTime: number = Math.round(
+    moment.duration(pauseTime.diff(execution.latestStartTime)).asSeconds()
+  );
+  console.log(
+    "timerFunction.addElapsedTime: Contagem a ser adicionada: ",
+    diffTime
+  );
+  let newElapsedTime: number = execution.elapsedTime + diffTime;
+  console.log(
+    "timerFunction.addElapsedTime: Pausando execução - Contagem atual: ",
+    newElapsedTime
+  );
+  execution.elapsedTime = newElapsedTime;
+  execution.currentState = ExecutionStatus.Paused;
+  return execution;
+};
+
+const getSingleOngoingExecution = async (id: number) => {
+  let lstExecution = await getOngoingExecutions();
+  if (!lstExecution || lstExecution.length === 0) {
+    console.warn("Não há execuções.");
+    return null;
+  }
+
+  if (JSON.parse(lstExecution).length > id) {
+    let execution: OngoingExecution = JSON.parse(lstExecution)[id];
```

Comment:
```suggestion
    const execution: OngoingExecution = JSON.parse(lstExecution)[id];
```

Considere verificar se há outras variáveis que nunca são alteradas no seu contexto, se tiver mais, elas devem ser `const` ao invés de `let`;